src/doors/TrainDoor.cpp

```diff
@@ -0,0 +1,502 @@
+                return;
+            }
+
+            if (param_state >= 2 || param_state <= 0) {
+                UtilityFunctions::push_error("[MaSzyna::Doors] State parameter (argument #2) is out of range");
+                return;
+            }
+
+            mover->OperateDoors(static_cast<side>(param_side), param_state, _range_t);
+            return;
+        }
+        UtilityFunctions::push_error("[MaSzyna::Doors] Unknown command: " + command);
+    }
+
+    void TrainDoor::_do_update_internal_mover(TMoverParameters *mover) {
+        const std::map<std::string, control_t>::iterator open_method_lookup =
+                door_controls.find(int_too_door_controls[door_open_method]);
+        mover->Doors.open_control =
+                open_method_lookup != door_controls.end() ? open_method_lookup->second : control_t::passenger;
+        const std::map<std::string, control_t>::iterator close_method_lookup =
+                door_controls.find(int_too_door_controls[door_close_method]);
+        mover->Doors.close_control =
+                close_method_lookup != door_controls.end() ? close_method_lookup->second : control_t::passenger;
+        mover->Doors.auto_duration = door_open_time;
+        mover->Doors.auto_velocity = door_auto_close_enabled ? door_auto_close_velocity : -1.0f;
+        mover->Doors.auto_include_remote = door_auto_close_remote;
+        mover->Doors.permit_needed = door_needs_permit;
+        mover->Doors.permit_presets.clear();
+        for (int i = 0; i < door_permit_list.size(); i++) {
+            if (door_permit_list[i] != Variant()) {
+                mover->Doors.permit_presets.emplace_back(static_cast<int>(door_permit_list[i]));
+            }
+        }
+
+        if (!mover->Doors.permit_presets.empty()) {
+            mover->Doors.permit_preset = door_permit_list_default;
+            mover->Doors.permit_preset =
+                    std::min<int>(mover->Doors.permit_presets.size(), mover->Doors.permit_preset) - 1;
+        }
+        mover->Doors.open_rate = open_speed;
+        mover->Doors.open_delay = door_open_delay;
+        mover->Doors.close_rate = close_speed;
+        mover->Doors.close_delay = door_close_delay;
+        mover->Doors.range = door_max_shift;
+        mover->Doors.range_out = door_max_shift_plug;
+        const std::map<std::string, int>::iterator lookup = door_types.find(int_to_door_types[door_open_method]);
+        mover->Doors.type =
+            lookup != door_types.end() ?
+                lookup->second :
+                2;
+        mover->Doors.has_warning = door_closure_warning;
+        mover->Doors.has_autowarning = auto_door_closure_warning;
+        mover->Doors.has_lock = door_blocked;
+
+        bool const remote_door_control {
+                mover->Doors.open_control == control_t::driver ||
+                mover->Doors.open_control == control_t::conductor || mover->Doors.open_control == control_t::mixed };
+        if (door_voltage == -1.0) {
```

Comment:
Nie podoba mi się ten -1 dla undefined. A -2? Albo -3? Lub -0.5? Ktoś może wstawić cokolwiek i to nie zadziała jak należy. 

IMO float dla tego propsa nie pasuje. Plik charakterystyki wyraźnie opisuje trzy możliwe wartości: `DoorVoltage=[0] (0 / 12 / 110)`. LoadFIZ na defaulcie ustawia 24 gdy jest remote door control a nie określono DoorVoltage (nie wiem czy w Maszynie 0 się do tego zalicza czy nie)